guest_os_booting: cover loadparm on interface

smitterl opened the pull request:
Depends on https://github.com/avocado-framework/avocado-vt/pull/4239

s390x supports more than one network boot entry since QEMU 10.1.0. The loadparm value can be used to boot into entries directly.

Add test case that confirms that the second network entry is booted into.

The test only covers that the right entry is accessed, not the actual installation. As such, the test case creates a directory per entry but uses the same bootable data. The message confirms that the right folder is accessed.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Multi-entry TFTP/PXE boot menus with selectable default and per-entry kernel/initrd handling.
  * Configurable list of TFTP boot entries for flexible boot content generation.
  * s390x virtio loadparm boot path support with prompt/expected-output checks.

* **Tests**
  * Extended boot-order tests to cover multi-entry menus and the loadparm scenario.

* **Chores**
  * Minor formatting/style tweaks and improved network resource cleanup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

liang-cong-red-hat commented on provider/virtual_network/tftpboot.py:
```diff
@@ -27,7 +27,32 @@
+    return "\n".join(contents)
+
+
+def create_tftp_content(install_tree_url, kickstart_url, arch, entries):
```
I found a script tp-libvirt/virttools/tests/src/virt_install/pxe_installation.py also refer to this function, I think this file also should be modified accordingly.

liang-cong-red-hat reviewed (approved):
Just one comments, others LGTM

smitterl commented:
> @liang-cong-red-hat spotted a backward compatibility issue that I need to fix so putting this in Draft shortly so it won't be merged. But given the already existing approvals, I think I'll merge once I confirmed I addressed it.

Additionally, I identified a test config issue which I also fixed. With the two additional commits:

```
JOB ID     : c753d9f6914b5e9b12af6291b3b59d5d83948375
JOB LOG    : /var/log/avocado/job-results/job-2025-09-26T05.51-c753d9f/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virt_install.installation.pxe_installation: STARTED
 (1/1) type_specific.io-github-autotest-libvirt.virt_install.installation.pxe_installation:  PASS (279.64 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/log/avocado/job-results/job-2025-09-26T05.51-c753d9f/results.html
JOB TIME   : 281.07 s
...
JOB ID     : 56e1e2f9c679ffa96bfa5af3f7afc6cb17bf23a1                                                                                                                                      
JOB LOG    : /var/log/avocado/job-results/job-2025-09-26T05.40-56e1e2f/job.log                                                                                                             
 (1/2) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.network.network_bootable.default: STARTED                                     
 (1/2) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.network.network_bootable.default:  PASS (67.25 s)                             
 (2/2) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.network.network_bootable.with_loadparm: STARTED                               
 (2/2) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.network.network_bootable.with_loadparm:  PASS (36.10 s)                       
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0                                                                                                          
JOB HTML   : /var/log/avocado/job-results/job-2025-09-26T05.40-56e1e2f/results.html                                                                                                        
JOB TIME   : 104.89 s    
```
